ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Update.hs

```diff
@@ -204,36 +203,31 @@ pureTryAddTx
 pureTryAddTx cfg txSize wti tx is values
   | let curSize = msNumBytes $ isMempoolSize is
   , curSize < getMempoolCapacityBytes (isCapacity is)
-  = -- We add the transaction if there is at least one byte free left in the mempool.
-  case eVtx of
-      -- We only extended the ValidationResult with a single transaction
-      -- ('tx'). So if it's not in 'vrInvalid', it must be in 'vrNewValid'.
+  = -- We add the transaction if there is at least one byte free left in the
+    -- mempool.
+  Processed $ case eVtx of
       Right vtx ->
-        assert (isJust (vrNewValid vr)) $
-          Processed $ TransactionProcessingResult
-            (Just is')
-            (MempoolTxAdded vtx)
-            (TraceMempoolAddedTx
-              vtx
-              (isMempoolSize is)
-              (isMempoolSize is')
-            )
+        TransactionProcessingResult
+          (Just is')
+          (MempoolTxAdded vtx)
+          (TraceMempoolAddedTx
+            vtx
+            (isMempoolSize is)
+            (isMempoolSize is')
+          )
       Left err ->
-        assert (isNothing (vrNewValid vr))  $
-          assert (length (vrInvalid vr) == 1) $
```

Comment:
Note to reviewer: Only one transaction is applied, this either fails of succeeds, but there is no need for this assert here.